### PR TITLE
account_invoice_from_picking initial commit

### PR DESCRIPTION
**Pull request by rvalyi**
This module allows to create invoice lines from picking orders.

A usage of this module is it allows you to create partial invoices for partial receipts/deliveries in case you forgot to create them after each receipt/delivery. Care has been taken for not creating discrepancies with the normal invoices created from the sale orders or pulling lines from the purchase order.

This module might also be required in some federal countries such as Brazil were you may produce invoice-like documents (a bit like European intrastat but with accounting entries too) when you simply move goods from one federal state to another without any sale or purchase order.

**This a work in progress** as there is still a bug: you can still pull lines from pickings related to paid sale orders. And tests are still missing.

### CODE REVIEW

**Comment by mbcosta** [comment]
I included the dependency of module onchange_helper ( to get the dictionary created by onchanges methods ), but the module has a problem with o2m fields, solved in this PR https://github.com/OCA/server-tools/pull/1468 . So this PR depends of the fix on onchange_helper module first.

cc @rvalyi @renatonlima @mileo

**Comment by mileo** [comment]
Some fixes: https://github.com/akretion/account-invoicing/pull/6

**Comment by mileo** [comment]
Hi, i detect some issue when insert the product 

https://www.youtube.com/watch?v=tHHJkgsMBco

**Review by mileo — CHANGES_REQUESTED** [review]
See video

**Comment by AaronHForgeFlow** [comment]
From my side this works well, I created a fix for a minor issue I found https://github.com/akretion/account-invoicing/pull/7

As this is PR is quite old, also a rebase would be good.

Thanks

UPDATE: I found several problems, qty invoiced in SO is not updated when invoicing from the picking and got error in views when creating the invoice from the action menu.

File: account_invoice_from_picking/README.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+#. Go to ...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
+.. branch is "8.0" for example
+
+Known issues / Roadmap
```
Comment (nikul-serpentcs) [inline account_invoice_from_picking/README.rst:43]: @rvalyi  If not any  Roadmap content then remove it.

File: account_invoice_from_picking/README.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+Images
+------
+
+* Odoo Community Association: `Icon <https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.svg>`_.
+
+Contributors
+------------
+
+* Firstname Lastname <email.address@example.org>
```
Comment (nikul-serpentcs) [inline account_invoice_from_picking/README.rst:67]: @rvalyi Can you please add your name

File: account_invoice_from_picking/README.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+* Second Person <second.person@example.org>
+
+Funders
+-------
+
+The development of this module has been financially supported by:
+
+* Company 1 name
```
Comment (nikul-serpentcs) [inline account_invoice_from_picking/README.rst:75]: ?

File: account_invoice_from_picking/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Account Invoice From Picking',
+    'summary': """
+        Allows to create invoice lines from picking orders""",
+    'version': '10.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'Akretion,Odoo Community Association (OCA)',
+    'website': 'www.akretion.com',
```
Comment (nikul-serpentcs) [inline account_invoice_from_picking/__manifest__.py:12]: URL should be `https://github.com/OCA/account-invoicing`

File: account_invoice_from_picking/models/account_invoice.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+from odoo import api, fields, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    picking_id = fields.Many2one(
+        'stock.picking',
+        string='Add Picking Order',
```
Comment (nikul-serpentcs) [inline account_invoice_from_picking/models/account_invoice.py:14]:
```suggestion
        string="Picking Order"
```

File: account_invoice_from_picking/models/stock.py [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+    def action_invoice_create(
+            self, active_ids, journal_id, group=False, type='out_invoice'):
+        """ Creates invoice based on the invoice state selected for picking.
+        @param journal_id: Id of journal
+        @param group: Whether to create a group invoice or not
+        @param type: Type invoice to be created
+        @return: Ids of created invoices for the pickings
+        """
+        todo = {}
+        for picking_id in active_ids:
```
Comment (nikul-serpentcs) [inline account_invoice_from_picking/models/stock.py:163]:
```suggestion
        for picking_id in self.env['stock.picking'].browse(active_ids):
```

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
nitpicking

**Comment by mbcosta** [comment]
Hi @mileo @aheficent @nikul-serpentcs thanks for reviews

At 3 weeks ago we noticed that there is other PR in the repo with same purpose https://github.com/OCA/account-invoicing/pull/509 , so I think the first thing to do should be choose one of them or try to join the best parts of the both code.

cc @rvalyi @renatonlima

**Comment by rvalyi** [comment]
superseded by https://github.com/OCA/account-invoicing/pull/509

**Pull request closed without merging**